server/design/toolmetrics/design.go

```diff
@@ -312,3 +377,162 @@ var ToolExecutionLog = Type("ToolExecutionLog", func() {
 		"function_id",
 	)
 })
+
+var TelemetryFilter = Type("TelemetryFilter", func() {
+	Description("Filter criteria for telemetry queries")
+
+	Attribute("from", String, "Start time in ISO 8601 format (e.g., '2025-12-19T10:00:00Z')", func() {
+		Format(FormatDateTime)
+		Example("2025-12-19T10:00:00Z")
+	})
+	Attribute("to", String, "End time in ISO 8601 format (e.g., '2025-12-19T11:00:00Z')", func() {
+		Format(FormatDateTime)
+		Example("2025-12-19T11:00:00Z")
+	})
+	Attribute("deployment_id", String, "Deployment ID filter", func() {
+		Format(FormatUUID)
+	})
+	Attribute("function_id", String, "Function ID filter", func() {
+		Format(FormatUUID)
+	})
+	Attribute("gram_urn", String, "Gram URN filter")
+})
+
+var SearchLogsFilter = Type("SearchLogsFilter", func() {
+	Description("Filter criteria for searching logs")
+
+	Extend(TelemetryFilter)
+
+	Attribute("trace_id", String, "Trace ID filter (32 hex characters)", func() {
+		Pattern("^[a-f0-9]{32}$")
+	})
+	Attribute("severity_text", String, "Severity level filter", func() {
+		Enum("DEBUG", "INFO", "WARN", "ERROR", "FATAL")
+	})
+	Attribute("http_status_code", Int32, "HTTP status code filter")
+	Attribute("http_route", String, "HTTP route filter")
+	Attribute("http_method", String, "HTTP method filter", func() {
+		Enum("GET", "POST", "PUT", "PATCH", "DELETE", "HEAD", "OPTIONS")
+	})
+	Attribute("service_name", String, "Service name filter")
+})
+
+var SearchLogsPayload = Type("SearchLogsPayload", func() {
+	Description("Payload for searching unified telemetry logs following OpenTelemetry Logs Data Model")
+
+	Attribute("filter", SearchLogsFilter, "Filter criteria for the search")
+	Attribute("cursor", String, "Cursor for pagination (UUID)", func() {
+		Format(FormatUUID)
+	})
```

Comment:
same as my other remark. treat it as an opaque string with no meaning to the client. don't provide hints about it's format.